source/brailleDisplayDrivers/albatross.py

```diff
@@ -0,0 +1,437 @@
+			"review_top": ("br(albatross):home1", "br(albatross):home2",),
+			"review_bottom": ("br(albatross):end1", "br(albatross):end2",),
+			"navigatorObject_toFocus": ("br(albatross):eCursor1", "br(albatross):eCursor2",),
+			"braille_toFocus": ("br(albatross):cursor1", "br(albatross):cursor2",),
+			"moveMouseToNavigatorObject": ("br(albatross):home1+home2",),
+			"moveNavigatorObjectToMouse": ("br(albatross):end1+end2",),
+			"navigatorObject_moveFocus": ("br(albatross):eCursor1+eCursor2",),
+			"braille_toggleTether": ("br(albatross):cursor1+cursor2",),
+			"braille_previousLine": ("br(albatross):up1", "br(albatross):up2", "br(albatross):up3",),
+			"braille_nextLine": ("br(albatross):down1", "br(albatross):down2", "br(albatross):down3",),
+			"braille_scrollBack": ("br(albatross):left", "br(albatross):lWheelLeft", "br(albatross):rWheelLeft",),
+			"braille_scrollForward": (
+				"br(albatross):right", "br(albatross):lWheelRight", "br(albatross):rWheelRight",),
+			"braille_routeTo": ("br(albatross):routing",),
+			"braille_reportFormatting": ("br(albatross):secondRouting",),
+			"braille_toggleFocusContextPresentation": ("br(albatross):attribute1+attribute3",),
+			"speechMode": ("br(albatross):attribute2+attribute4",),
+			"reviewMode_previous": ("br(albatross):f1",),
+			"reviewMode_next": ("br(albatross):f2",),
+			"navigatorObject_parent": ("br(albatross):f3",),
+			"navigatorObject_firstChild": ("br(albatross):f4",),
+			"navigatorObject_previous": ("br(albatross):f5",),
+			"navigatorObject_next": ("br(albatross):f6",),
+			"navigatorObject_current": ("br(albatross):f7",),
+			"navigatorObject_currentDimensions": ("br(albatross):f8",),
+			"review_activate": ("br(albatross):f7+f8",),
+			"dateTime": ("br(albatross):f9",),
+			"say_battery_status": ("br(albatross):f10",),
+			"title": ("br(albatross):f11",),
+			"reportStatusLine": ("br(albatross):f12",),
+			"reportCurrentLine": ("br(albatross):f13",),
+			"sayAll": ("br(albatross):f14",),
+			"review_currentCharacter": ("br(albatross):f15",),
+			"review_currentLine": ("br(albatross):f16",),
+			"review_currentWord": ("br(albatross):f15+f16",),
+			"review_previousLine": ("br(albatross):lWheelUp", "br(albatross):rWheelUp",),
+			"review_nextLine": ("br(albatross):lWheelDown", "br(albatross):rWheelDown",),
+			"kb:windows+d": ("br(albatross):attribute1"),
+			"kb:windows+e": ("br(albatross):attribute2"),
+			"kb:windows+b": ("br(albatross):attribute3"),
+			"kb:windows+i": ("br(albatross):attribute4"),
+		},
+	})
+
+
+class InputGestureKeys(braille.BrailleDisplayGesture):
+
+	source = BrailleDisplayDriver.name
+
+	def __init__(self, keys):
+		super().__init__()
+		self.keyCodes = set(keys)
+		names = []
+		for key in self.keyCodes:
+			if 2 <= key <= 41 or 111 <= key <= 150:
+				names.append("routing")
+				self.routingIndex = self._getRoutingIndex(key)
+			elif 43 <= key <= 82 or 152 <= key <= 191:
+				names.append("secondRouting")
+				self.routingIndex = self._getRoutingIndex(key)

```

Comment:
This leaves the logic for `_getRoutingindex` spread out. There is high coupling and low encapsulation.

Instead consider:
```python

	_routingRanges = {  # range values are inclusive
		(2, 41): (2, "routing"),
		(43, 82): (43, "secondRouting"),
		(111, 150): (71, "routing"),
		(152, 191): (112, "secondRouting"),

	def _getRoutingIndex(self, key: int) -> Optional[Tuple[str, int]]:
		""" Get the routing index, if the key is in a routing index range, returns the name of the range and the
		index ??Within that range??.
		See L{_routingRanges}
		"""
		for range, val in _routingRanges:
			rangeStart, rangeEnd = range
			indexOffset, routingName = value
			if rangeStart <= key <= rangeEnd:
				return routingName, key - indexOffset
		return None
```
Initially, I didn't expect the offset values to be required. I thought the `rangeStart` value was used in all cases. The current code obfuscates that some other offset is used in two cases.
A comment about how to calculate the offset would help readers, better would be the explicit calculation from the initial range values.